File: encryptDecrypt.cs
using System;
using System.Web;
using System.Security.Cryptography;

public class TestEncrypt
{
	
	const String tokenKey = "x8FWQUedjyiUGlTf5appPQ==";
	const String appCode = "DEMOAPP";
	const String companyCode = "10";
	const String version = "2";
	const String method = "getStudents";
	const String endPoint = "http://tass.school.edu.au/tassweb/api/";
	
	public static void Main()
	{
		string original = "{\"currentstatus\":\"current\"}";
		
		var UTF8 = new System.Text.UTF8Encoding();
		var encString = Encrypt( original );
		
		Console.WriteLine("Original:   {0}", original );
		Console.WriteLine("Encrypted:  {0}", encString );
		Console.WriteLine("Decrypted:  {0}", Decrypt( encString ) );
		Console.WriteLine("URL:  {0}?method={1}&appcode={2}&company={3}&v={4}&token={5}", endPoint , method , appCode , companyCode , version , HttpUtility.UrlEncode ( encString ) );
		
	}

 	public static string Encrypt (string toEncrypt)
    {
        var UTF8 = new System.Text.UTF8Encoding();
		
		// The Token Key is already Base64.
		byte[] keyArray   	  = Convert.FromBase64String(tokenKey);
		
		// The Paramater String is UTF8.
        byte[] toEncryptArray = UTF8.GetBytes(toEncrypt);

        RijndaelManaged rDel  = new RijndaelManaged();
        rDel.Key              = keyArray;
        rDel.Mode             = CipherMode.ECB;
        rDel.BlockSize        = 128;
        rDel.Padding   		  = PaddingMode.PKCS7;

        ICryptoTransform cTransform  = rDel.CreateEncryptor( keyArray , null );

        byte[] resultArray    = cTransform.TransformFinalBlock(toEncryptArray, 0, toEncryptArray.Length);
		
		return Convert.ToBase64String(resultArray, 0, resultArray.Length);
    }

    public static string Decrypt (string toDecrypt)
    {
        var UTF8 = new System.Text.UTF8Encoding();
		
		// The Token Key is already Base64.
		byte[] keyArray  	  = Convert.FromBase64String(tokenKey);
		
		// The encrypted string is already Base64.
        byte[] toEncryptArray = Convert.FromBase64String(toDecrypt);

        RijndaelManaged rDel  = new RijndaelManaged();
        rDel.Key              = keyArray;
        rDel.Mode             = CipherMode.ECB;
        rDel.BlockSize        = 128;
        rDel.Padding    	  = PaddingMode.PKCS7;
	
        ICryptoTransform cTransform = rDel.CreateDecryptor( keyArray , null );

        byte[] resultArray    = cTransform.TransformFinalBlock(toEncryptArray, 0, toEncryptArray.Length);

        return UTF8.GetString(resultArray);
    }	
}

File: encryptDecrypt.js
'use strict';
const https = require('https');
const crypto = require('crypto');

/* 

    Special thansk to Ben Nadel
    https://www.bennadel.com/blog/3118-using-aes-ecb-pkcs5padding-encryption-in-coldfusion-and-decrypting-values-in-node-js.htm

*/

const tokenKey      = "x8FWQUedjyiUGlTf5appPQ==";
const appCode       = "DEMOAPP";
const companyCode   = "10";
const version       = "2";
const method        = "getStudents";
const endPoint      = "https://api.tassweb.com.au/tassweb/api/index.cfm";
let parameterString = '{"currentstatus":"current"}';


var encrypt = function( parameterString , tokenKey ) {

    var binaryEncryptionKey = new Buffer.from( tokenKey, "base64" );

    var cipher = crypto.createCipheriv( "AES-128-ECB", binaryEncryptionKey, Buffer.alloc(0) );

    var encryptedString = (
        cipher.update( parameterString, "utf8", "base64" ) +
        cipher.final( "base64" )
    );

    return encryptedString;
};

var decrypt = function( encryptedInput , tokenKey ) {

    var binaryEncryptionKey = new Buffer.from( tokenKey, "base64" );
    var binaryIV = new Buffer.from( '' );

    var decipher = crypto.createDecipheriv( "AES-128-ECB", binaryEncryptionKey, binaryIV );

    // When decrypting we're converting the Base64 input to UTF-8 output.
    var decryptedString = (
        decipher.update( encryptedInput, "base64", "utf8" ) +
        decipher.final( "utf8" )
    );

    return decryptedString;
};

var constructRequest = function() {

    var token = encodeURIComponent( encrypt ( parameterString , tokenKey ) );

    return endPoint + '?appcode=' + appCode +'&v=' + version + '&method=' + method + '&company=' + companyCode + '&token=' + token; 

};

var makeAPIRequest = function(){

    var requestURL = constructRequest();

    https.get( requestURL , (resp) => {

        if ( resp.statusCode === 200 ) {
        
            let data = '';

            resp.on('data', (chunk) => {
                data += chunk;
            });

            resp.on('end', () => {

                let searchResponse = JSON.parse(data);

                console.log( searchResponse );

            });

        } else {

            console.log( "Error: " + resp.statusCode );

        }

    }).on("error", (err) => {

        console.log("Error: " + err.message);

    });

}
console.log( 'Original Paramater String : ' + parameterString );
console.log( 'Encrypted Paramater String : ' + encrypt ( parameterString , tokenKey ) );
console.log( 'Decrypted Paramater String : ' + decrypt ( encrypt ( parameterString , tokenKey ) , tokenKey ) );
console.log( 'Request URL : ' + constructRequest() );

makeAPIRequest();

